### PR TITLE
Cb 153 modify sortmerna module to use prebuilt index

### PR DESCRIPTION
**Pull request by lzholl**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🧑‍💻 Code Refactor
- [x] ✅ Test

## Description

This PR accomplishes a few different things:
1. I modified the sortmerna module to skip indexing and use the provided SILVA db index by default. 
2. I was able to test the parameter files that I created for Mru & Mgo and adjust them as needed.
3. The STAR & salmon reference indices are now built for both genomes, so I can move them to our s3 bucket.

I configured the parameters file to skip the STAR & salmon steps for now so that I could get the indices built first. Basically, the pipeline preps the indices from the provided genome files, runs through the pre-processing steps of the pipeline, and skips the alignment & downstream steps. 

## Jira Task
 
Covers both [CB-150](https://arkeabio.atlassian.net/browse/CB-150?atlOrigin=eyJpIjoiZThkM2JjZjU5NWU2NDQ4ZThlYzY2ODRmOTg1YjkyNGIiLCJwIjoiaiJ9) and [CB-153](https://arkeabio.atlassian.net/browse/CB-153?atlOrigin=eyJpIjoiZDUwNDM0ZTQxNGRjNDQwMjg4Nzc3OGM4ZjQwYzM0OTUiLCJwIjoiaiJ9).

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📕 notebook(s)
- [x] 🙅 no documentation needed

## Code Quality
- [x] 🎯 The code is properly formatted and adheres to the project's style guide
- [x] 🎯 The code runs without any errors or exceptions

## Testing

To test, make sure the arkeabio-rnaseq-mru.json (parameter file) is in the **working directory** and mru_samplesheet.csv & reads are in the **localdata** folder. For this test I just used single end reads from an Mru RNA-Seq experiment. 

You should also have a **localdata/genomes** folder that contains four files: 2 gtf and 2 fasta. 

You'll also need the SILVA index files found on our s3 bucket [here](https://arkeabio-nextflow-general.s3.amazonaws.com/meta-t_reference/idx/). Place the four files in the folder **localdata/silvadb/idx**. 

Finally, the actual SILVA reference sequences can be found [here](https://arkeabio-nextflow-general.s3.amazonaws.com/meta-t_reference/smr_v4.3_default_db.fasta). Place this file in the folder **localdata/silvadb**.

Test it out with: 

`nextflow run main.nf -profile docker -params-file arkeabio-rnaseq-mru.json`

11 of the 15 processes should run.

edit to add that you should run it on the sandbox.

### CODE REVIEW

File: nextflow.config
Comment (tfalkarkea) [inline nextflow.config:242]: Did this test file get implemented or is it for later?

**Review by tfalkarkea — APPROVED** [review]
Glad to see another pipeline coming together!

A few general points I want to mention:
1. We already discussed, but mentioning that this needs to be run on the sandbox would have saved me some time in getting set up.
2. Further, if you already have these files on the server then it really helps out the pull reviewer to have these stored in the shared `/data/` folder instead of needing to run around and download and place them where necessary. It would be the most streamlined to have the reference files in a shared directory that are also referenced in a test file so the reviewer can run the pipeline right away.
3. I would generally discourage including multiple Jira tasks in one pull review, and to split them among different branches to keep track of tasks more easily.

Changes themselves look good, just some overhead I wanted to comment on!

**Comment by tfalkarkea** [comment]
Also I believe you'll need to give the GitHub team Bioinformatics write access to get the approval to go through.

File: nextflow.config
Comment (lzholl) [reply nextflow.config:242]: It's for later. I don't remember even adding this tbh...

**Comment by lzholl** [comment]
Thanks. I've given write access to the team. I'll keep the other things in mind for future PRs. I forgot about the shared `/data/` directory and I guess mentioning the sandbox in general, even though that's where I ran all the tests. :expressionless:

**Pull request merged**
